src/components/Pill/src/Pill.vue

```diff
@@ -0,0 +1,90 @@
+<template>
+	<span
+		:class="$s.Pill"
+		:style="style"
+		v-bind="$attrs"
+		v-on="$listeners"
+	>
+		<!-- @slot pill content -->
+		<slot />
+	</span>
+</template>
+
+<script>
+import cssValidator from '@square/maker/utils/css-validator';
+import { MThemeKey, defaultTheme, resolveThemeableProps } from '@square/maker/components/Theme';
+
+/**
+ * @inheritAttrs div
+ * @inheritListeners div
+ */
+export default {
+	inject: {
+		theme: {
+			default: defaultTheme(),
+			from: MThemeKey,
+		},
+	},
+
+	inheritAttrs: false,
+
+	props: {
+		/**
+		 * pattern defined at theme level
+		 */
+		pattern: {
+			type: String,
+			default: undefined,
+		},
+		/**
+		 * text color, also border color if no bg color
+		 */
+		textColor: {
+			type: String,
+			default: undefined,
+			validator: cssValidator('color'),
+		},
+		/**
+		 * bg & border color
+		 */
+		bgColor: {
+			type: String,
+			default: undefined,
+			validator: cssValidator('color'),
+		},
+	},
+
+	computed: {
+		...resolveThemeableProps('pill', [
+			'pattern',
+			'textColor',
+			'bgColor',
+		]),
+		style() {
+			const textColor = this.resolvedTextColor || 'var(--maker-color-neutral-90, #1b1b1b)';
```

Comment:
Can we move the fallback into the styles below? I think that's what we're doing for most components - defining resolved color vars in the js and fallbacks in the css.